sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java

```diff
@@ -69,6 +71,7 @@
   protected MessageType fileSchema;
   protected MessageType requestedSchema;
   protected StructType sparkSchema;
+  protected ParsedVersion writerVersion;
```

Comment:
Maybe it's worth adding some comments on why we track `writerVersion` here. It's a bit confusing when reading this.